### PR TITLE
Add Elastic Fleet reset script

### PR DESCRIPTION
**Pull request by defensivedepth**

### CODE REVIEW

File: salt/manager/tools/sbin_jinja/so-elastic-fleet-reset [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+sed -i "/fleet_grid_enrollment_token_general.*/d" /opt/so/saltstack/local/pillar/global/soc_global.sls
+sed -i "/fleet_grid_enrollment_token_heavy.*/d" /opt/so/saltstack/local/pillar/global/soc_global.sls
+
+printf "\n\nDeleting Elastic Fleet data...\n\n"
+
+ALIASES=".fleet-servers .fleet-policies-leader .fleet-agents .fleet-artifacts .fleet-enrollment-api-keys .kibana_ingest"
+for ALIAS in ${ALIASES}
+do
+    # Get all concrete indices from alias
+    INDXS=$(curl -K /opt/so/conf/kibana/curl.config -s -k -L -H "Content-Type: application/json" "https://localhost:9200/_resolve/index/${ALIAS}"  | jq -r '.aliases[].indices[]')
```
Comment (jertel) [inline salt/manager/tools/sbin_jinja/so-elastic-fleet-reset:43]: If Kibana fails to respond, what happens?

File: salt/manager/tools/sbin_jinja/so-elastic-fleet-reset [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    # Delete all resolved indices
+    for INDX in ${INDXS}
+    do
+            printf "\nDeleting $INDX \n"
+            curl -K /opt/so/conf/kibana/curl.config -s -k -L -H "Content-Type: application/json" "https://localhost:9200/${INDX}" -XDELETE
```
Comment (jertel) [inline salt/manager/tools/sbin_jinja/so-elastic-fleet-reset:49]: Handle Kibana response error

**Review by jertel — CHANGES_REQUESTED** [review]

File: salt/manager/tools/sbin_jinja/so-elastic-fleet-reset [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+# Inform user we are about to remove Elastic Fleet data
+echo
+echo "This script will remove the current Elastic Fleet install & all of its data and rerun Elastic Fleet setup."
+echo
+echo "If you would like to proceed, please type "AGREE" and hit ENTER."
+echo
+# Read user input
+read INPUT
+if [ "$INPUT" != "AGREE" ] ; then exit 0; fi
```
[the remark this answers is not in the document]
Comment (defensivedepth) [reply salt/manager/tools/sbin_jinja/so-elastic-fleet-reset:23]: Completed

File: salt/manager/tools/sbin_jinja/so-elastic-fleet-reset [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+
+printf  "\nUninstalling all Elastic Agents on all Grid Nodes...\n\n"
+salt \* cmd.run "elastic-agent uninstall -f" queue=True
```
[the remark this answers is not in the document]
Comment (defensivedepth) [reply salt/manager/tools/sbin_jinja/so-elastic-fleet-reset:27]: It's expected that there may be errors here if Elastic Agent is not actually installed - and that's fine.

**Review by jertel — APPROVED** [review]

**Pull request merged**